### DISCUSSION TITLE
Error when building - Make sure to set `TAURI_PRIVATE_KEY` environment variable.

### DISCUSSION THREAD

**Discussion by privatedev11** [tauri-apps/tauri#6013] [Q&A]
When I build my app, I get this error.
```A public key has been found, but no private key. Make sure to set `TAURI_PRIVATE_KEY` environment variable.```
It lets me continue, however.
I set the private key as an environment variable in Windows to rule it off, but it didn't work.
How do I fix this?

**Comment by neatfx** [comment]
tauri.conf.json > tauri > updater > active > false

**Comment by markkorput** [comment]
(still) having the same problem, is there any solution that does not require disabling the updater functionality?

Update: I get this error on `macOS 14.2.1` using `tauri-cli-build 1.5.7` trying to create a build for macOS.

**Comment by xavierskip** [comment]
if you don't want updater, you can try tauri.conf.json > bundle> [targets](https://v2.tauri.app/reference/config/#targets) remove updater

"targets": ["deb", "appimage", "nsis", "app", "dmg","updater"],
 ↓
"targets": ["deb", "appimage", "nsis", "app", "dmg"],

**Comment by ZaidSunasra** [comment]
To overcome this problem:
Open a new powershell in projects root
type the following command: $env:TAURI_SIGNING_PRIVATE_KEY="Your private key" 
type: $env:TAURI_SIGNING_PRIVATE_KEY  It should return your private key then you are good to go.
Now in the same powershell type: npm run tauri build (or any of your package manager).
Actually env variables do not persist across terminals. I faced the same problem and this helped me to overcome it.
